index.js

```diff
@@ -23,11 +23,23 @@ function useComponentSize(ref) {
 
   useLayoutEffect(() => {
     handleResize()
-    window.addEventListener('resize', handleResize)
 
-    return () => {
-      window.removeEventListener('resize', handleResize)
+    if (ResizeObserver) {
+      const resizeObserver = new ResizeObserver(() => handleResize())
```

Comment:
this throw an error since you later try to set a `const` to a new value

I think you want to use `let` here:

```suggestion
      let resizeObserver = new ResizeObserver(() => handleResize())
```